src/printer.js

```diff
@@ -1557,8 +1558,11 @@ function genericPrintNoParens(path, options, print) {
       return "string";
     case "DeclareTypeAlias":
     case "TypeAlias": {
-      const parent = path.getParentNode(1);
-      if (parent && parent.type === "DeclareModule") {
+      const nodeSource = getNodeSource(n, options.originalText)
```

Comment:
this function is super expensive as it does a copy of the entire file in an array line by line. Instead you could do

```js
options.originalText.slice(0, locStart(n)).match(/declare\s*$/)
```